flink-cdc-e2e-tests/flink-cdc-pipeline-e2e-tests/src/test/java/org/apache/flink/cdc/pipeline/tests/MysqlE2eITCase.java

```diff
@@ -333,16 +335,32 @@ public void testSchemaChangeEvents() throws Exception {
     }
 
     @Test
-    public void testDroppingTable() throws Exception {
-        Thread.sleep(5000);
-        LOG.info("Sleep 5 seconds to distinguish initial DDL events with dropping table events...");
-        long ddlTimestamp = System.currentTimeMillis();
-        Thread.sleep(5000);
-        LOG.info("Going to drop tables after timestamp {}", ddlTimestamp);
+    public void testDanglingDropTableEventInBinlog() throws Exception {
+        // Create a new table for later deletion
+        try (Connection connection = mysqlInventoryDatabase.getJdbcConnection();
+                Statement statement = connection.createStatement()) {
+            statement.execute("CREATE TABLE live_fast(ID INT PRIMARY KEY);");
+        }
+
+        String logFileName = null;
+        Long logPosition = null;
+
+        try (Connection connection = mysqlInventoryDatabase.getJdbcConnection();
+                Statement statement = connection.createStatement()) {
+            ResultSet rs = statement.executeQuery("SHOW BINARY LOGS;");
```

Comment:
IIUC we can add a note: one more transaction to push the binlog file latest position move forward ?